tests/files/tests.py

```diff
@@ -169,6 +169,27 @@ def test_io_wrapper(self):
             test_file.seek(0)
             self.assertEqual(test_file.read(), (content * 2).encode())
 
+    def test_file_lock(self):
+        file_path = Path(__file__).parent / 'test.png'
+        f1 = open(file_path)
+        f2 = open(file_path)
```

Comment:
These files should be explicitly closed using a `with`. Right now, they result in errors in the console. Same in `test_file_shared_lock`:

```
Exception ignored in: <_io.FileIO name='/home/jon/devel/django/tests/files/test.png' mode='rb' closefd=True>
Traceback (most recent call last):
  File "/usr/lib64/python3.9/unittest/case.py", line 550, in _callTestMethod
    method()
ResourceWarning: unclosed file <_io.TextIOWrapper name='/home/jon/devel/django/tests/files/test.png' mode='r' encoding='UTF-8'>
Exception ignored in: <_io.FileIO name='/home/jon/devel/django/tests/files/test.png' mode='rb' closefd=True>
Traceback (most recent call last):
  File "/usr/lib64/python3.9/unittest/case.py", line 550, in _callTestMethod
    method()
ResourceWarning: unclosed file <_io.TextIOWrapper name='/home/jon/devel/django/tests/files/test.png' mode='r' encoding='UTF-8'>
Exception ignored in: <_io.FileIO name='/home/jon/devel/django/tests/files/test.png' mode='rb' closefd=True>
Traceback (most recent call last):
.  File "/usr/lib64/python3.9/unittest/case.py", line 550, in _callTestMethod
.........    method()
ResourceWarning: unclosed file <_io.TextIOWrapper name='/home/jon/devel/django/tests/files/test.png' mode='r' encoding='UTF-8'>
Exception ignored in: <_io.FileIO name='/home/jon/devel/django/tests/files/test.png' mode='rb' closefd=True>
Traceback (most recent call last):
  File "/usr/lib64/python3.9/unittest/case.py", line 550, in _callTestMethod
    method()
ResourceWarning: unclosed file <_io.TextIOWrapper name='/home/jon/devel/django/tests/files/test.png' mode='r' encoding='UTF-8'>
.................................
----------------------------------------------------------------------
Ran 43 tests in 0.046s
```

Reply:
Yes, please use `self. addCleanup()`.